docs/doc_index.py

```diff
@@ -0,0 +1,163 @@
+    global doc_return_value
+    doc_return_value = Doc(title, options, entries)
+
+
+def example_entry(name, operator_ref = None):
+    """Place given notebook or doc page in the toctree and optionally add a reference from operator
+    documentation to that notebook or page.
+
+    Parameters
+    ----------
+    name : str
+        Name of jupyter notebook or rst file, must contain proper extension.
+    operator_ref : OpReference or List[OpReference], optional
+        Optional reference, defined by `op_reference()` call, by default None
+    """
+    return ExampleEntry(name, operator_ref)
+
+
+def op_reference(operator, docstring):
+    """Add a reference from operator to this notebook with specified docstring.
+
+    Parameters
+    ----------
+    operator : str
+        Name of operator without nvidia.dali prefix, for example fn.resize or fn.gaussian_blur
+    docstring : str
+        Text that would appear in the see also block for given link.
+    """
+    return OpReference(operator, docstring)
+
+
+def _obtain_doc(py_file):
+    """Extract the doc() definition from index.py file"""
+    with open(py_file, 'r') as f:
+        doc_file = f.read()
+        exec(doc_file)
+        return doc_return_value
+
+
+def _document_examples(path, result_dict={}):
+    py_file = path + ".py"
+    rst_file = path + ".rst"
+
+    doc_contents = _obtain_doc(py_file)
+    tab = " " * 3
+    with open(rst_file, "w") as f:
+        if isinstance(doc_contents.title, str):
+            f.write(f".. title:: {doc_contents.title}\n\n")
+        else:
+            title, level = doc_contents.title
+            f.write(f"{title}\n{level * len(title)}\n\n")
+
+        f.write(f".. toctree::\n")
+        if not isinstance(doc_contents.options, list):
+            doc_contents.options = [doc_contents.options]
+        for option in doc_contents.options:
+            f.write(f"{tab}{option}\n")
+        f.write("\n")
+        for entry in doc_contents.entries:
+            f.write(f"{tab}{entry}\n")
+
+    canonical_path = Path(path)
+    base_path = canonical_path.parent
+    for entry in doc_contents.entries:
+        if isinstance(entry, str):
+            _document_examples(str(base_path / entry), result_dict)
+            # TODO(klecki) if someone wants to link to a index page from operator
+            # it can be kinda added here
+        else:
+            if entry.operator_ref is None:
+                continue
+            # Ternary expression is super long and Python formatters are abysmal
+            if isinstance(entry.operator_ref, OpReference):
+                op_refs = [entry.operator_ref]
+            else:
+                op_refs = entry.operator_ref
+
+            for op_ref in op_refs:
+                if not op_ref.operator in result_dict:
+                    result_dict[op_ref.operator] = []
+
+                result_dict[op_ref.operator].append(
+                    (op_ref.docstring, str(base_path / entry.name)[:-6] + ".html"))
```

Comment:
Can this be a method of `ExampleEntry`?

Reply:
`ExampleEntry` doesn't track it's base URL, so I would prefer to keep it here.

Reply:
Sure